### PR TITLE
Creating a new gem for background jobs

### PR DESCRIPTION
**Pull request by solojavier**
This extracts the behaviour that was part of wowzer to create background jobs.

This will be reused for proteus and so..

### CODE REVIEW

**Comment by solojavier** [comment]
@andresf and @guilleart could you take a look ?

**Comment by andresf** [comment]
Where is the handling of the queue-specific values when queueing a job?

**Comment by andresf** [comment]
Said in another way, the way this is done right now will make properties of a job object clash with the properties of the mongo queue object.

**Comment by solojavier** [comment]
@andresf What about now?

**Comment by andresf** [comment]
You only took out the retry options from the queue properties...that doesn't solve the problem.

Please take a look at how it was originally implemented, as what's here is not how it worked.

**Comment by solojavier** [comment]
@andresf There you go, change the name and also added the queue_options

**Comment by andresf** [comment]
Need a test for job handler classes that are namespaced (e.g. Test::JobHandler)

File: lib/soter/job_worker.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+   
+      while(job = @queue.lock_next(process_id))
+        log "#{process_id}: Starting work on job #{job['_id']}"
+        log "#{process_id}: Job info =>"
+
+        job.each {
+          |key, value| log  "#{process_id}: {#{key} : #{value}}" }
+          
+          begin
+            job_handler = eval(job['handler_class']).new(job['job_params'])
```
Comment (andresf) [inline lib/soter/job_worker.rb]: Don't use eval.

http://vemod.net/module-recursive_const_get

**Comment by solojavier** [comment]
@andresf done with your comments :octopus:

**Comment by solojavier** [comment]
@andresf :shipit: ??

**Comment by solojavier** [comment]
#4

**Comment by andresf** [comment]
+1 after the above is taken care of

**Pull request merged**